### DISCUSSION TITLE
Python Not Working Properly

### DISCUSSION THREAD

**Discussion by sha4bab** [ZI-Youtuber/ZI-Youtuber#4] [Q&A] [answered]
hi, I see you are a python coder but I have problems whenever I run the code no window shows up only if I use the python turtle it shows up. can you help me pls maybe it's because I'm using a mac but I think that's not the reason, not even my terminal works! i use pycharm/ visual studio code

**Comment by zoraezimran** [comment]
Ok can you sned me a screen shot of the chat in scratch we were doing so I can just remember what I was talking about

**Reply by sha4bab** [reply]
ok

**Reply by zoraezimran** [reply]
https://github.com/Scratch-Dev-Team I have sent an invite to the team just go to the link then just accept the invitation

**Reply by sha4bab** [reply]
ok

**Comment by zoraezimran** [comment]
@sha4bab hmm... if you are using visual studio it is a bad idea to use because it doesn't work for me properly either so I use pycharm for python type of stuff and it is best to get the latest version of it so it works better

**Reply by sha4bab** [reply]
ok but pycharm is limited and even that doesn't work properly for me

**Reply by zoraezimran** [reply]
@sha4bab Like I said use the latest version and just use that for python and you can do the rest with visual studio because that is what I do

**Reply by sha4bab** [reply]
I do but still, it only lets me use turtle graphics

**Reply by zoraezimran** [reply]
@sha4bab hmm.. that's odd

**Reply by zoraezimran** [reply]
@sha4bab Wait let me think about it

**Reply by sha4bab** [reply]
is there any other platform you reccomend?

**Reply by zoraezimran** [reply]
@sha4bab like you mean code editor?

**Reply by sha4bab** [reply]
i mean like the python editor

**Reply by sha4bab** [reply]
i tried almost all of the python editors none works I installed all the packages but still

**Reply by zoraezimran** [reply]
@sha4bab yeah I'm thinking about it and also I just realised that you were using a mac :) and not Windows 10 :)

**Reply by zoraezimran** [reply]
> i tried almost all of the python editors none works I installed all the packages but still

yeah I am thinking about it

**Reply by sha4bab** [reply]
ye but i have windows 10 operating system installed and shouldn't it work for a mac?

**Reply by zoraezimran** [reply]
@sha4bab yeah it should be I am thinking but it says pycharm is suitable for mac. Also use your Windows 10 computer instead of your mac for pychram

**Reply by sha4bab** [reply]
and my terminal doesn't work

**Reply by zoraezimran** [reply]
@sha4bab terminal is the main problem and previewing as well

**Reply by sha4bab** [reply]
i don't have a windows 10 computer I have a mac and have windows 10 installed

**Reply by zoraezimran** [reply]
@sha4bab Like in virtual machine?

**Reply by sha4bab** [reply]
?

**Reply by zoraezimran** [reply]
ok no problem you might not know that but How can Windows 10 run on a mac?

**Reply by sha4bab** [reply]
oh no i can install windows 10 on my computer even if its a mac

**Reply by sha4bab** [reply]
you can buy windows 10 for a mac

**Reply by zoraezimran** [reply]
@sha4bab that's really cool to be honest. Ok let me think about a suitable code editor for you

**Reply by zoraezimran** [reply]
> you can buy windows 10 for a mac

oh cool

**Reply by sha4bab** [reply]
ok

**Reply by zoraezimran** [reply]
@sha4bab maybe I could talk to you tomorrow about this and find a suitable code editor for you to use. Just keep on looking here whenever you can just in case I send an answer

**Reply by sha4bab** [reply]
thanks so much for the help tho

**Reply by zoraezimran** [reply]
@sha4bab no problem! Always here for a muslim brother or anyone who needs help

**Reply by sha4bab** [reply]
thx!!!

**Reply by sha4bab** [reply]
I also used atom and it was pretty much like vscode

**Reply by sha4bab** [reply]
I mean the python turtle is useful but it's so limited compared to the python language

**Comment by zoraezimran** [comment, marked as answer]
@sha4bab The only things I could find were pycharm and replit. Replit is a browser based code editor and works really well but the terminal doesn't work on it. Pycharm is the best python code editor. If any more problems tell me

**Reply by sha4bab** [reply]
oh thx sorry for the late reply and thx for the help I really appreciate it

**Reply by zoraezimran** [reply]
@sha4bab no prob
